General updates

Jeroen0494 opened the pull request:
Hi there,

More general updates.
* I replaces all shell instances with the new tunable shells_path for uniformity.
* Lot's of KDE updates
* New profiles for kconsole and cron-ntp
* Add new tunable hci_id

There are many, many ,many more updates needed for profiles. Especially dbus rules will require significant updates across all applications.

roddhjav commented on apparmor.d/groups/kde/kded5:
```diff
@@ -29,16 +30,59 @@ profile kded5 @{exec_path} {
 
   @{bin}/kcminit            rPx,
   @{bin}/pgrep              rCx -> pgrep,
+  @{bin}/python{,**}        rix,
```
```
@{bin}/python3.@{int}
```

roddhjav commented on apparmor.d/groups/kde/konsole:
```diff
@@ -0,0 +1,59 @@
+
+  owner @{user_share_dirs}/konsole/{,**} rw,
+
+  owner @{run}/user/@{uid}/bus rw,
```
Use `include <abstractions/bus-session>` instead.

roddhjav commented on apparmor.d/groups/kde/konsole:
```diff
@@ -0,0 +1,59 @@
+
+include <tunables/global>
+
+@{exec_path} = /{usr/,}bin/konsole
```
Use: `@{bin}/konsole`

roddhjav commented on apparmor.d/groups/kde/konsole:
```diff
@@ -0,0 +1,59 @@
+  include <abstractions/dri>
+  include <abstractions/fonts>
+  include <abstractions/freedesktop.org>
+  include <abstractions/kde-strict>
```
`kde-strict` already has `fonts` and `freedesktop.org` so you can remove them.

roddhjav commented on apparmor.d/groups/kde/konsole:
```diff
@@ -0,0 +1,59 @@
+
+  /usr/share/color-schemes/{,**} r,
+  /usr/share/konsole/{,**} r,
+  /usr/share/hwdata/pnp.ids r,
```
Present in `kde-strict`

roddhjav commented on apparmor.d/groups/kde/konsole:
```diff
@@ -0,0 +1,59 @@
+  include <abstractions/freedesktop.org>
+  include <abstractions/kde-strict>
+  include <abstractions/nameservice-strict>
+  include <abstractions/ubuntu-konsole>
```
Do no use this abs. (it includes `kde`)

roddhjav commented on apparmor.d/groups/kde/konsole:
```diff
@@ -0,0 +1,59 @@
+  ptrace (read),
+
+  @{exec_path} mr,
+  @{bin}/** rPUx,
+  @{user_bin_dirs} rPUx,
```
Is it needed?

Terminal app ususally only need to start a shell unconfined. Like
```
  @{bin}/@{shells}      rUx,
```

roddhjav commented on apparmor.d/groups/kde/kded5:
```diff
@@ -14,6 +14,7 @@ profile kded5 @{exec_path} {
   include <abstractions/bus/org.bluez>
   include <abstractions/consoles>
   include <abstractions/dconf-write>
+  include <abstractions/dbus-network-manager-strict>
```
Only use dbus abs from this project (in `bus/`)

Jeroen0494 replied:
I'll come back to this after some testing. I don't remember which dbus rules exactly are required, so I'll run it without this include first.

roddhjav replied:
Most likelly you need: `include <abstractions/bus/org.freedesktop.NetworkManager>`

roddhjav commented:
I am wondering if all the shells should be allowed in this case:
- Allowing all shell from `@{shells}` make sense for interactive shells, so that the user can use its own shell.
- However, in this context it is a shell used not interactively by program. I would rather limit this to sh, bash and dash.

It means you replacement could kept but renamed using a new shell tunable like:
```
@{sh} = sh zsh bash dash
@{sh_path} = @{bin}/@{sh}
```

Jeroen0494 commented:
Hi, thank your very much for the review.

I started writing the konsole profile before the kde-strict profile was created, so there was some overlap.

I'll admit I'm not exactly great at writing dbus rules. The part I mostly struggle with is the peer names and labels. The aa-log command helps a lot in this regard. Pointer are welcome!

Regarding the shells, it depends on the system default. On Arch linux, as long as the shell is sh compatible it can be used. Ubuntu is a bit more fussy when it comes to the default shell. I'll leave the decision for shell selection up to you.

roddhjav commented on apparmor.d/abstractions/bus/org.freedesktop.UPower:
```diff
@@ -12,13 +12,9 @@
        peer=(name=:*, label=upowerd),
 
-  dbus send bus=system path=/org/freedesktop/UPower/devices/DisplayDevice
+  dbus send bus=system path=/org/freedesktop/UPower/devices/*
```
That does not work:
- `peer=(name=org.freedesktop.UPower, label=upowerd),` and `peer=(name=:*, label=upowerd),` are two different thing, you cannot merge them
- Be verbose in this bus file: add a new entry only for `PropertiesChanged` (with both peer name and label)

roddhjav commented on apparmor.d/abstractions/bus/org.freedesktop.PackageKit:
```diff
@@ -2,6 +2,41 @@
+       member=PropertiesChanged
+       peer=(name=:*),
+
+  dbus send bus=system path=/org/freedesktop/DBus
+       interface=org.freedesktop.DBus
+       member={AddMatch,GetNameOwner,Hello,RequestName}
+       peer=(name=:*),
```
Remove:
1. Not the place for this
2. Already included in `bus-system`
3. Always specify the target program profile with the peer label

Jeroen0494 replied:
Because parts of KDE aren't confined yet, and some parts are started via shell and run unconfined, some peer labels are `unconfined`. What do I do in such a situation?

roddhjav commented on apparmor.d/abstractions/bus/org.freedesktop.PackageKit:
```diff
@@ -2,6 +2,41 @@
+       member=Introspect
+       peer=(name=:*),
+
+  dbus send bus=system path=/@{int}@{int}@{int}@{int}@{int}_edcedacc
+       interface=org.freedesktop.PackageKit.Transaction
+       member={Destroy,Finished,Package}
+       peer=(name=:*),
```
Remove this. It gives to much access use a dbus directive in the profile that needs it.

See: https://apparmor.pujol.io/development/dbus/#dbus-abstractions

roddhjav commented on apparmor.d/abstractions/bus/org.freedesktop.PackageKit:
```diff
@@ -2,6 +2,41 @@
+  dbus receive bus=system path=/@{int}@{int}@{int}@{int}@{int}_edcedacc
+       interface=org.freedesktop.PackageKit.Transaction
+       member=GetUpdates
+       peer=(name=:*),
```
You **must** specify the peer label: this is one of the value added by this project: as we have profile for everything, we can restrict the dbus further setting the peer label (the profile name). 

*(same everywhere you did not mention peer label)*

roddhjav commented on apparmor.d/groups/kde/kauth-backlighthelper:
```diff
@@ -21,8 +23,11 @@ profile kauth-backlighthelper @{exec_path} {
   @{sys}/class/backlight/ r,
   @{sys}/class/leds/ r,
   @{sys}/devices/@{pci}/backlight/**/brightness rw,
+  @{sys}/devices/@{pci_bus}/**/drm/card@{int}/**/enabled r,
```
Use `@{pci}/` instead of `/@{pci_bus}/**/`

roddhjav commented on apparmor.d/groups/kde/kcminit:
```diff
@@ -12,6 +12,8 @@ profile kcminit @{exec_path} {
   include <abstractions/gtk>
   include <abstractions/kde-strict>
 
+  unix (connect, receive, send) type=stream peer=(addr="@/tmp/.X11-unix/X[0-9]*"),
```
That should already be included in `X-strict` (that in included by `kde-strict`)

roddhjav commented on apparmor.d/groups/kde/kded5:
```diff
@@ -29,16 +30,59 @@ profile kded5 @{exec_path} {
 
   signal (send) set=hup peer=xsettingsd,
 
+  dbus receive bus=system path=/org/freedesktop/NetworkManager/SecretAgent
```
In general: you need both peer name & label

roddhjav commented on apparmor.d/groups/kde/kded5:
```diff
@@ -29,16 +30,59 @@ profile kded5 @{exec_path} {
+       interface=org.freedesktop.PolicyKit1{,.*}
+       peer=(label=polkitd),
+
+  dbus receive bus=system path=/org/bluez/hci@{int}/@{hci_id}{,/**}
+       interface=org.freedesktop.DBus.Properties
+       member=PropertiesChanged
+       peer=(label=bluetoothd),
```
Remove, use `abstractions/bus/org.bluez`

roddhjav commented on apparmor.d/groups/kde/kded5:
```diff
@@ -29,16 +30,59 @@ profile kded5 @{exec_path} {
+       member={PropertiesChanged,AccessPointAdded,AccessPointRemoved}
+       peer=(label=NetworkManager),
+
+  dbus receive bus=system path=/org/freedesktop/PolicyKit1{,/**}
```
Try to use `abstractions/bus/org.freedesktop.PolicyKit1` instead, only add rule here if it requires more (or even give full talk access: https://apparmor.pujol.io/development/dbus/#dbus-directive

roddhjav commented on apparmor.d/groups/kde/kstart:
```diff
@@ -8,17 +8,29 @@ abi <abi/3.0>,
+  owner @{user_share_dirs}/kservices5/ r,
+  owner @{user_share_dirs}/kservices5/ServiceMenus/ r,
+
+  owner @{run}/user/@{uid}/at-spi/bus_0 rw,
```
Use `abstractions/bus-accessibility` instead.

roddhjav commented on apparmor.d/groups/kde/kstart:
```diff
@@ -8,17 +8,29 @@ abi <abi/3.0>,
+  /var/lib/flatpak/exports/share/mime/ r,
+
+  owner @{user_cache_dirs}/mesa_shader_cache/index rw,
+  owner @{user_config_dirs}/kwinrc r,
```
Already in `kde-strict`

roddhjav commented on apparmor.d/groups/kde/kstart:
```diff
@@ -8,17 +8,29 @@ abi <abi/3.0>,
+
+  owner @{run}/user/@{uid}/at-spi/bus_0 rw,
+
+  @{sys}/devices/@{pci_bus}/**/{device,revision,subsystem_device,subsystem_vendor,uevent,vendor} r,
```
use the `dri` abstraction instead.

roddhjav commented on apparmor.d/groups/network/nm-online:
```diff
@@ -16,8 +16,23 @@ profile nm-online @{exec_path} {
        member=StateChanged
        peer=(name=:*, label=NetworkManager),
+  
+  dbus send bus=system path=/org/freedesktop/DBus
```
This is already included in `bus-system`

roddhjav commented on apparmor.d/groups/network/rpcbind:
```diff
@@ -0,0 +1,16 @@
+
+include <tunables/global>
+
+@{exec_path} = /{,user/}{,s}bin/rpcbind
```
@{bin}/rpcbind

roddhjav commented on apparmor.d/profiles-a-f/flatpak:
```diff
@@ -10,6 +10,8 @@ include <tunables/global>
 profile flatpak @{exec_path} flags=(attach_disconnected,mediate_deleted,complain) {
   include <abstractions/base>
   include <abstractions/consoles>
+  include <abstractions/dbus-session-strict>
```
Use the dbus abs from this profile (they have to good label): `abstractions/bus-session`

roddhjav commented on apparmor.d/profiles-m-r/mdevctl:
```diff
@@ -11,6 +11,11 @@ profile mdevctl @{exec_path} {
   include <abstractions/base>
 
   @{exec_path} mr,
+  @{bin}/basename rix,
+  @{bin}/find rix,
+  @{bin}/getopt rix,
+  @{bin}/jq   rix,
+  @{bin}/sort rix,
```
mdevctl is https://github.com/mdevctl/mdevctl. This is not a bash script. Why is this here?

Jeroen0494 replied:
I can't find the original audit logs for this anymore, and a system reboot doesn't produce them either. I'll just remove this for now.

roddhjav commented:
What distribution / DE are you using?

Jeroen0494 commented:
I'm using KDE neon. Quite a bit of dbus rules are coming up with peer labels unconfined.

Thanks for another review, I'll work in fixing your point on Wednesday. I missed the move to abstractions/bus/* so I'll fix those as well.

roddhjav commented:
> Regarding the shells, it depends on the system default. On Arch linux, as long as the shell is sh compatible it can be used. Ubuntu is a bit more fussy when it comes to the default shell. I'll leave the decision for shell selection up to you.

What do you mean by fussy?
Anyway, having a separate `@{sh}` variable allow someone with it own settings to add any special and future shells, while beeing way more restrictive.

roddhjav commented:
Forgot to ask, what your GPU? That may change some stuff regarding `dri` stuff ?

Jeroen0494 commented:
> > Regarding the shells, it depends on the system default. On Arch linux, as long as the shell is sh compatible it can be used. Ubuntu is a bit more fussy when it comes to the default shell. I'll leave the decision for shell selection up to you.
> 
> What do you mean by fussy?

Peculiar, specific, change it and break it. I believe Ubuntu includes some bash-ism in it's scripts, maybe even some dash-ism. Arch Linux avoids this and sticks with sh.

> Anyway, having a separate `@{sh}` variable allow someone with it own settings to add any special and future shells, while being way more restrictive.

Alright, I'll work on changing this in my PR.

roddhjav commented:
Ok, so when you have resource specific for interl GPU (drm, mesa), it needs to be seen has any possible gpu. And you need to use `include <abstractions/graphics>`

Jeroen0494 commented:
So, before I start working on my dbus rules, I have a question. The following are generated:

```
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.DBus send bus=system path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=Hello peer_label=dbus-daemon
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.DBus send bus=system path=/ interface=org.freedesktop.DBus member=AddMatch peer_label=dbus-daemon
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.DBus send bus=system path=/ interface=org.freedesktop.DBus member=GetNameOwner peer_label=dbus-daemon
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.DBus send bus=system path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch peer_label=dbus-daemon
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.DBus send bus=system path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch peer_label=dbus-daemon
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.DBus send bus=system path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameHasOwner peer_label=dbus-daemon
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.DBus send bus=system path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=GetNameOwner peer_label=dbus-daemon
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager interface=org.freedesktop.DBus.Properties member=GetAll peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager interface=org.freedesktop.NetworkManager member=GetDevices peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings interface=org.freedesktop.NetworkManager.Settings member=ListConnections peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/ActiveConnection/4 interface=org.freedesktop.DBus.Properties member=GetAll peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Devices/3 interface=org.freedesktop.DBus.Properties member=GetAll peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/ActiveConnection/2 interface=org.freedesktop.DBus.Properties member=GetAll peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Devices/2 interface=org.freedesktop.DBus.Properties member=GetAll peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/ActiveConnection/1 interface=org.freedesktop.DBus.Properties member=GetAll peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Devices/8 interface=org.freedesktop.DBus.Properties member=GetAll peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/1 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/2 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/3 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/4 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/5 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/6 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/7 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/8 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/9 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/10 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/11 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Devices/1 interface=org.freedesktop.DBus.Properties member=GetAll peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Devices/7 interface=org.freedesktop.DBus.Properties member=GetAll peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/12 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/13 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/14 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/15 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/16 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
ALLOWED flatpak-app//&xdg-dbus-proxy dbus_method_call org.freedesktop.NetworkManager send bus=system path=/org/freedesktop/NetworkManager/Settings/17 interface=org.freedesktop.NetworkManager.Settings.Connection member=GetSettings peer_label=NetworkManager
```

The first dbus-daemon rules are already present, because dbus-strict is included in the xdg-dbus-proxy profile. The other ones are present by `abstractions/bus/org.freedesktop.NetworkManager`. So why are there rules still generated?

roddhjav commented:
Try adding `abstractions/bus/org.freedesktop.NetworkManager` and `bus-system` into`flatpak-app`. As the profile is stacked, the real confined profile is the inclusion of `flatpak-app` & `xdg-dbus-proxy`.

Jeroen0494 commented:
I've closed the reviews as resolved where I've provided fixes. For the others, my audit logs are so full I cannot find the rules anymore.

I'll work on creating a new shell variable.

I can squash my commits when everything is ready for merge.

roddhjav commented:
Nice, tell me when you think it is ready to be merged.

Jeroen0494 commented:
Okay, it took a while, but I cleaned up my commits. The new shell variable is now in a separate commit. Everything else is in general update.

Now I need to look into mdevctl and kded5.

Jeroen0494 commented:
It seems some profile transitions isn't happening correctly for Flatpak:
```
ALLOWED flatpak-app link owner @{run}/user/@{uid}/akregatorLKFYUI.221.kioworker.socket -> @{run}/user/@{uid}/#971 comm=akregator requested_mask=l denied_mask=l
ALLOWED flatpak-app open @{PROC}/sys/kernel/core_pattern comm=kioslave5 requested_mask=r denied_mask=r
ALLOWED flatpak-app connect @{run}/systemd/resolve/io.systemd.Resolve comm=kioslave5 requested_mask=wr denied_mask=wr
ALLOWED flatpak-app open @{PROC}/sys/kernel/core_pattern comm=kio_http_cache_ requested_mask=r denied_mask=r
ALLOWED flatpak-app connect @{run}/dbus/system_bus_socket comm=pool requested_mask=wr denied_mask=wr
```

I've added a profile transition to kioslave5, but I'm not sure what's going on here.
flatpak-app already has dbus-system, I've added it to flatpak because it's still denied, I'm guessing for the same reason as the other one above.

roddhjav commented:
`flatpak-app` does not have `bus-system` (only `bus-session`), you can add it (but in `abstractions/bwrap-app`)

Jeroen0494 commented:
Okay, ready for another review round.

Jeroen0494 commented:
And I'm guessing in the meantime abstraction/dbus-session was renamed to abstraction/bus-session.

roddhjav commented:
No, this is not a renaming:
1. `abstraction/dbus-session` is provided by `apparmor-profiles` and give full access to the session bus.
2. `abstraction/bus-session` is new (December 2023) from the full rewrite of dbus rule in this project. It is the up to date equivalent of `abstraction/dbus-session-strict` with peer label coresponding to the profile name defined in `apparmor.d`

So, when you still have `dbus-session` you cannot replace it by `bus-session`.

Jeroen0494 commented:
Okay, fixed.

And I've added multiarch to lib, thanks to kioslave5:

ALLOWED dolphin exec @{lib}/@{multiarch}/libexec/kf5/kioslave5 -> @{lib}/@{multiarch}/libexec/kf5/kioslave5 info="profile transition not found" comm=dolphin requested_mask=x denied_mask=x error=-13

Jeroen0494 commented:
Okay, I'm done. Please just fix the tunable for me, I'm getting extremely frustrated. You can commit to this branch.

roddhjav commented on apparmor.d/groups/kde/dolphin:
```diff
@@ -74,7 +80,9 @@ profile dolphin @{exec_path} {
   owner @{PROC}/@{pid}/mountinfo r,
   owner @{PROC}/@{pid}/mounts r,
 
+  @{sys}/devices/virtual/block/dm-0/uevent r,
```
Use `@{sys}/devices/virtual/block/dm-@{int}/uevent` instead.

roddhjav commented on apparmor.d/abstractions/bus/org.freedesktop.UPower:
```diff
@@ -15,10 +15,11 @@
-  dbus send bus=system path=/org/freedesktop/UPower/devices/DisplayDevice
+       peer=(name=:*, label=upowerd),
+
+  dbus receive bus=system path=/org/freedesktop/UPower/devices/{mouse_hidpp_battery_0,battery_BAT0}
```
`/org/freedesktop/UPower/devices/*` otherwise it depends of the device type too much.

roddhjav commented on apparmor.d/abstractions/bwrap-app:
```diff
@@ -10,7 +10,8 @@
   include <abstractions/audio>
   include <abstractions/consoles>
-  include <abstractions/dbus-session>
+  include <abstractions/bus-session>
+  include <abstractions/bus-system>
```
You need to the rules:
```
dbus bus=session,
dbus bus=system,
```

Otherwise it will break a lot of things. We may improve this later.

roddhjav commented on apparmor.d/groups/kde/konsole:
```diff
@@ -0,0 +1,67 @@
+  include <abstractions/dri>
+  include <abstractions/kde-strict>
+  include <abstractions/nameservice-strict>
+  include <abstractions/ubuntu-xterm>
```
Do not use this abstraction, it is too large (it even includes `<abstractions/kde>`)

Jeroen0494 replied:
I thought I had already removed it... Must have been something to do with rebasing / squashing.

roddhjav commented on apparmor.d/groups/kde/konsole:
```diff
@@ -0,0 +1,67 @@
+  owner @{user_config_dirs}/konsolerc.@{rand6} rwl -> @{user_config_dirs}/#@{int},
+
+  owner @{user_cache_dirs}/icon-cache.kcache rw,
+  owner @{user_cache_dirs}/mesa_shader_cache/index rw,
```
Remove the followin:
- `@{user_cache_dirs}/mesa_shader_cache/index rw,`
- `@{PROC}/sys/dev/i915/perf_stream_paranoid r,`

And add the `mesa` abstraction.

roddhjav commented:
> And I've added multiarch to lib, thanks to kioslave5:
>
> ALLOWED dolphin exec @{lib}/@{multiarch}/libexec/kf5/kioslave5 -> @{lib}/@{multiarch}/libexec/kf5/kioslave5 info="profile transition not found" comm=dolphin requested_mask=x denied_mask=x error=-13

What you are proposing (`/{,usr/}{,@{multiarch}/}lib{,exec,32,64}`) would not cover this path (you need `@{lib}=/{,usr/}lib{,exec,32,64}/{,@{multiarch}/}`. But even this way, this is program specific and therefore I would rather not have it for every program.

Jeroen0494 commented:
I think that's it? I have a lot more fixes, but will create a new PR for this.

roddhjav commented:
Thanks, that is a very good contribution. Merged!
